modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -932,14 +932,30 @@ private void processValueAttributeOfOperation(SCIMResourceTypeSchema schema, Pat
     }
 
     private List<Map<String, String>> transformMembersAttributeToMap(MultiValuedAttribute multiValuedMembersAttribute)
-            throws CharonException {
+            throws CharonException, BadRequestException {
 
         List<Map<String, String>> memberList = new ArrayList<>();
         List<Attribute> subValuesList = multiValuedMembersAttribute.getAttributeValues();
         for (Attribute subValue : subValuesList) {
             ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
             Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
 
+            // Check if `value` (member id) attribute is present and not empty.
+            if (!subAttributesList.containsKey(SCIMConstants.CommonSchemaConstants.VALUE)) {
+                throw new BadRequestException(ResponseCodeConstants.DESC_BAD_REQUEST,
+                        ResponseCodeConstants.INVALID_SYNTAX);
+            } else if (((SimpleAttribute)
+                    (subAttributesList.get(SCIMConstants.CommonSchemaConstants.VALUE))).getStringValue().isEmpty()) {
```

Comment:
Looks like the getStringValue() can return a null (https://github.com/wso2/charon/blob/734ad3bfcd18f83b1e8ccb48172a84071b821d74/modules/charon-core/src/main/java/org/wso2/charon3/core/attributes/SimpleAttribute.java#L77). Hence the isEmpty() check is prone to NPEs.

To avoid that, we could use StringUtils.isEmpty here

Reply:
Thank you for the feedback. 
I fixed it in the [985923e](https://github.com/wso2/charon/pull/387/commits/985923e7a1dd92ad8fb01825982c036a34d550e9)  commit.